src/StatsdClient/Worker/AsynchronousWorker.cs

```diff
@@ -77,10 +62,10 @@ void Dequeue()
 
                     if (_handler.OnIdle())
                     {
-                        Task.Delay(waitDuration).Wait();
+                        _waiter.Wait(waitDuration);
```

Comment:
We have a linear growth for the duration. This is not necessarily a proble, but wondering if we should consider an exponential increase in the wait time? Just a thought, I don't have an issue with this.

Reply:
Line 66 we double the waiting duration so this is an exponential increase. `waitDuration` is an `Timespan` object and it has `operator +` but not `operator *`.

Reply:
Yes, you're right, the waitDuration is not linear in growth! My bad! That mistake also showed in my other comment 🤦‍♂